### PR TITLE
Fix some type inference bugs

### PR DESCRIPTION
**Pull request by jmbockhorst**
Found some inference bugs while working on using type inference to show compiler diagnostics.

### CODE REVIEW

**Comment by Razzeee** [comment]
Seems to conflict, are there any test cases we should add?

**Review by Razzeee — CHANGES_REQUESTED** [review]
Looks good otherwise

**Comment by jmbockhorst** [comment]
Ready to merge.

File: src/util/treeUtils.ts [hunk 1]
```diff
@@ -2190,7 +2211,7 @@ export class TreeUtils {
             child.firstNamedChild?.text,
         ),
       )
-      .filter(Utils.notUndefined.bind(this))
+      .filter(Utils.notUndefined)
```
Comment (Razzeee) [inline src/util/treeUtils.ts:2214]: Is there a reason your removing this?

Comment (jmbockhorst) [reply src/util/treeUtils.ts:2214]: No, sorry probably happened in the merge.

Comment (Razzeee) [reply src/util/treeUtils.ts]: Readded

**Pull request merged**